src/components/route-book/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { useBook } from '@/fetchers/book';
+import { fromBaseUnit } from '@/old/utils/math/hiLo';
+import { Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import {
+  Position,
+  PositionId,
+} from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { Amount } from '@penumbra-zone/protobuf/penumbra/core/num/v1/num_pb';
+import { getDisplayDenomExponent } from '@penumbra-zone/getters/metadata';
+import { innerToBech32Address } from '@/old/utils/math/bech32';
+import { uint8ArrayToBase64 } from '@/old/utils/math/base64';
+import { JsonValue } from '@bufbuild/protobuf';
+import { useAssets } from '@/shared/state/assets';
+import { round } from '@/shared/round';
+import { useComputePositionId } from '@/shared/useComputePositionId';
+
+interface Route {
+  lpId: string;
+  position: Position;
+  price: number;
+  amount: number;
+}
+
+interface RouteWithTotal extends Route {
+  total: number;
+}
+
+function getTotals(data: Route[], isBuySide: boolean, limit: number): RouteWithTotal[] {
+  const totals: RouteWithTotal[] = Object.values(
+    data.reduce((displayData: Record<string, RouteWithTotal>, route: Route) => {
+      const key = route.price;
+      return {
+        ...displayData,
```

Comment:
thought: I wonder if this spread, creating a new object on every loop, can be inefficient on large data sets. Maybe not a concern. If so though, can do something like `const displayData = new Map<number, RouteWithTotal>();` with a normal for loop.